tools/pyi/gen_pyi.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import print_function
+import os
+import sys
+import inspect
+
+needed_modules = set()
+
+
+def type_to_python(typename):
+    typename = {'int64_t': 'int',
+                'Scalar': 'Union[float, int]',
+                'ScalarType': 'dtype',
+                'Device': 'Union[device, str, None]',
+                'IntList': 'Union[Tuple[int, ...], List[int, ...]]',
+                'Tensor?': 'Optional[Tensor]',
+                'Tensor': 'Tensor',
+                'bool': 'bool',
+                'double': 'float',
+                'Generator *': 'Generator',
+                'Generator*': 'Generator',
+                'std::vector<Tensor>': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'TensorList': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'Storage': 'Storage',  # inaccurate (FloatStorage...)
+                'SparseTensorRef': 'Tensor',
```

Comment:
@zdevito ...this type shouldn't show up in Declarations.yaml, right?

Reply:
probably not, but the handling of SparseTensor was pretty weird (at least at some point)

Reply:
I'll go through them and see which ones are actually used (I used to generate hints for all functons in the yaml before I moved to "import torch and see what's actually needed").